### DISCUSSION TITLE
How can i handle error when  fetch api failed in getStaticPaths

### DISCUSSION THREAD

**Discussion by wildfrontend** [vercel/next.js#20365] [Help] [answered]
if i get a error in getStaticPaths

how can i deal ?

it is deal in getStatisProps?

**Comment by jamesmosier** [comment, marked as answer]
To catch errors you could wrap your logic in a try/catch:

```js
try {
  // your logic
} catch (e) {
  // the error
}
```

Otherwise because `getStaticPaths` runs at build time you typically do not need to add this logic in because if `getStaticPaths` fails at build time you would want to know this and stop the execution of the build.
